src/callbacks_stage/subcell_bounds_check_2d.jl

```diff
@@ -8,12 +8,35 @@
 @inline function check_bounds(u, mesh::AbstractMesh{2}, equations, solver, cache,
                               limiter::SubcellLimiterIDP,
                               time, iter, output_directory, save_errors)
-    (; positivity) = solver.volume_integral.limiter
+    (; local_minmax, positivity) = solver.volume_integral.limiter
     (; variable_bounds) = limiter.cache.subcell_limiter_coefficients
     (; idp_bounds_delta) = limiter.cache
 
+    if local_minmax
+        for v in limiter.local_minmax_variables_cons
+            v_string = string(v)
+            key_min = Symbol(v_string, "_min")
+            key_max = Symbol(v_string, "_max")
+            deviation_min = idp_bounds_delta[key_min]
+            deviation_max = idp_bounds_delta[key_max]
+            for element in eachelement(solver, cache), j in eachnode(solver),
+                i in eachnode(solver)
+
+                var = u[v, i, j, element]
+                deviation_min[1] = max(deviation_min[1],
+                                       variable_bounds[key_min][i, j, element] - var)
+                deviation_max[1] = max(deviation_max[1],
+                                       var - variable_bounds[key_max][i, j, element])
+            end
```

Comment:
I believe that the loop over the elements should be threaded. Moreover, the current version does not seem to be thread-safe as all threads will try to read and write the variables `deviation_min` and `deviation_max`.

Reply:
This shows in the example `examples/tree_2d_dgsem/elixir_euler_blast_wave_sc_subcell_nonperiodic.jl`. With one thread, `check_bounds` takes about 5% of the computing time. With 10 threads, it takes about 20% of the computing time.

Reply:
Thanks, that's a good point.
But I actually think it would be cleaner to add the changes to a new PR. Then, I can request a review on the specific changes. Is that okay with you @amrueda, @sloede ?

Reply:
That's fine for me. We can work on improving the computational efficiency of the bounds check in a new PR. @sloede, if you are ok with it, then I think we are done :slightly_smiling_face:

Reply:
Yes, please use a separate PR :-)